From: Daniel Kiper <dkiper@net-space.pl>
Subject: xen: disable clock timer when shutting down.

This is needed as some Amazon EC2 machines can't restart the timer
properly in the kexec-ed kernel, as jiffies are not being incremented.

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

---
 arch/x86/xen/time.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/arch/x86/xen/time.c
+++ b/arch/x86/xen/time.c
@@ -289,7 +289,15 @@ static void xen_timerop_set_mode(enum cl
 
 	case CLOCK_EVT_MODE_UNUSED:
 	case CLOCK_EVT_MODE_SHUTDOWN:
+#if 0
+		/*
+		 * Disabled due to some Amazon EC2 machines are not able to
+		 * restart timer properly in crash kernel. If this happens then
+		 * it hangs in loop in calibrate_delay_direct() because jiffies
+		 * are not incremented.
+		 */
 		HYPERVISOR_set_timer_op(0);  /* cancel timeout */
+#endif
 		break;
 	}
 }
